### PR TITLE
Fix/frontend env replace

### PR DESCRIPTION
**Pull request by thoniTUB**
@awildturtok 
I found two mismatches with the `replace-env-at-runtime.sh`-script:
- Shebang pointed to `/bin/bash` which is not available in the runner image from the docker file
- `run_e2e_frontend.sh` provided an env-file as argument to the replace-script, whereas the Dockerfile provided the index.html that is to be replaced

The replace-script now takes two arguments. The first one is the index.html and is mandatory, the second  file is an env-file and is optional.

More over i added the `PROXY` build argument to the Dockerfile, so it is possible for yarn to pull dependencies behind a proxy.

### CODE REVIEW

File: frontend/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,14 @@
 FROM mhart/alpine-node:16 AS builder
 
+ARG PROXY
+
+# Set up proxy if necessary because yarn does not recognize the corresponding env variables
+RUN if [[ -z $PROXY ]]; then echo "Skip setting up proxy" ; else ( \
```
Comment (awildturtok) [inline frontend/Dockerfile:6]: you might want to check if yarn looks for lower or uppercase `HTTP_PROXY`. For some reason these both exist and arent universal

File: frontend/scripts/replace-env-at-runtime.sh [hunk 2]
```diff
@@ -1,20 +1,45 @@
...
+
+# Check for one or two arguments or exit 
+if  [ \( "$#" -lt 1 \) -o \( "$#" -gt 2 \) ] 
+then
+    echo "$usage"
+    exit 1
+fi
+
+# Check if env file is provided and readable
+if [ "$#" -eq 2 ]
```
Comment (awildturtok) [inline frontend/scripts/replace-env-at-runtime.sh:20]: Why isn't the env already sourced when invoking this file?

Comment (awildturtok) [reply frontend/scripts/replace-env-at-runtime.sh:20]: You might also want to just use `env` to invoke it

**Review by awildturtok — CHANGES_REQUESTED** [review]

File: frontend/scripts/replace-env-at-runtime.sh (hunk 2 above)
Comment (thoniTUB) [reply frontend/scripts/replace-env-at-runtime.sh:20]: but then, i would need to run the following command in a subshell

File: frontend/Dockerfile (hunk 1 above)
Comment (thoniTUB) [reply frontend/Dockerfile:6]: I just noticed, that using `docker build .` doesn't have problems with the proxy, but using `docker compose up --build` does 🤔

**Review by Kadrian — APPROVED** [review]
Tested locally using
```
docker build -t conquery .
docker run -e REACT_APP_API_URL=http://some-conquery-url --name conquery -p 8000:80 conquery
```

The replacement works for me

File: frontend/Dockerfile (hunk 1 above)
Comment (thoniTUB) [reply frontend/Dockerfile]: Habs wieder raus genommen, es geht irgendwie auch ohne wenn man `HTTP_PROXY` etc. als build arg von außen definiert

File: frontend/scripts/replace-env-at-runtime.sh (hunk 2 above)
Comment (thoniTUB) [reply frontend/scripts/replace-env-at-runtime.sh:20]: wobei man bei `env` nicht direkt ein env-File angeben kann

Comment (awildturtok) [reply frontend/scripts/replace-env-at-runtime.sh:20]: `env $(cat ./env) replace-env-at-runtime.sh`?

Comment (thoniTUB) [reply frontend/scripts/replace-env-at-runtime.sh:20]: Habs probiert, Dafür muss man die env-datei vorher säubern (Kommentare, vmtl auch zeilen umbrüche)

Comment (thoniTUB) [reply frontend/scripts/replace-env-at-runtime.sh:20]: So würde es gehen: `(source .env.e2e ; ./scripts/replace-env-at-runtime.sh build/index.html.test)`

Comment (awildturtok) [reply frontend/scripts/replace-env-at-runtime.sh:20]: Fixt du das noch?

**Review by awildturtok — APPROVED** [review]

File: frontend/scripts/replace-env-at-runtime.sh (hunk 2 above)
Comment (thoniTUB) [reply frontend/scripts/replace-env-at-runtime.sh]: Ich hatte es auf dem Server nochmal getestet und muss das sourcen des env-files mit `set -+a` umranden

**Pull request merged**